lib/ansible/modules/cloud/amazon/aws_kms.py

```diff
@@ -625,134 +607,173 @@ def compare_grants(existing_grants, desired_grants, purge_grants=False):
     return to_add, to_remove
 
 
-def ensure_enabled_disabled(connection, module, key):
-    changed = False
-    if key['key_state'] == 'Disabled' and module.params['enabled']:
-        try:
-            connection.enable_key(KeyId=key['key_arn'])
-            changed = True
-        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-            module.fail_json_aws(e, msg="Failed to enable key")
+def cancel_key_deletion(connection, module, key):
+    key_id = key['key_arn']
+    if key['key_state'] != 'PendingDeletion':
+        return False
 
-    if key['key_state'] == 'Enabled' and not module.params['enabled']:
-        try:
-            connection.disable_key(KeyId=key['key_arn'])
-            changed = True
-        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-            module.fail_json_aws(e, msg="Failed to disable key")
-    return changed
+    try:
+        connection.cancel_key_deletion(KeyId=key_id)
+        # key is disabled after deletion cancellation
+        # set this so that ensure_enabled_disabled works correctly
+        key['key_state'] = 'Disabled'
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Failed to cancel key deletion")
 
+    return True
 
-def update_alias(connection, module, key_id, alias):
-    if not alias.startswith('alias/'):
-        alias = 'alias/' + alias
-    aliases = get_kms_aliases_with_backoff(connection)['Aliases']
-    if key_id:
-        # We will only add new aliases, not rename existing ones
-        if alias not in [_alias['AliasName'] for _alias in aliases]:
+
+def ensure_enabled_disabled(connection, module, key, enabled):
+    desired_state = ('Disabled', 'Enabled')[enabled]
```

Comment:
Nice, I like how you did that

Reply:
I don't know, this seems too clever (and not hugely readable).

Prefer `desired_state = 'Enabled' if enabled else 'Disabled'`

Reply:
I would have preferred to us a ternary ``desired_state = enabled ? 'Enabled' : 'Disabled'`` but Python doesn't seem to have one, and this is what my Google search came up with as an alternative.

My understanding was that the inlined if statement was strongly discouraged in Python.

A friend once told me you have to be twice as clever to debug something as to write it.  If you're worried about how clever it is then it's not going to help with debugging in future, so I'll happily switch to something if based (I'd probably not inline the if though)

Reply:
Do you have any references for the inline if being discouraged?

If you don't like the inline if, then I'd go with the normal if block.

Reply:
I'll drop back to the normal if block.  Looking deeper I probably over-interpreted some of the comments about them being discouraged.  What I personally don't like about the Python 'ternary' operator is that there's an implicit ()
`desired_state = ('Enabled' if enabled else 'Disabled')` which isn't obvious to me.